pkg/daemon/update.go

```diff
@@ -158,14 +158,19 @@ func (dn *Daemon) catchIgnoreSIGTERM() {
 }
 
 // update the node to the provided node configuration.
-func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig) error {
+func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig) (retErr error) {
 	if dn.nodeWriter != nil {
-		if err := dn.nodeWriter.SetUpdateWorking(dn.kubeClient.CoreV1().Nodes(), dn.name); err != nil {
+		if err := dn.nodeWriter.SetUpdateWorking(dn.kubeClient.CoreV1().Nodes(), dn.nodeLister, dn.name); err != nil {
 			return err
 		}
 	}
 
 	dn.catchIgnoreSIGTERM()
+	defer func() {
+		if retErr != nil {
```

Comment:
Oh that's an interesting pattern, didn't realize it was that easy to do "do this if we're returning an error".  That's quite hard in other languages.